src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/CpsPackageReferenceProject.cs

```diff
@@ -202,68 +209,69 @@ private void IgnoreUnsupportProjectReference(PackageSpec project)
-                var fileInfo = new FileInfo(assetsFilePath);
-                IList<LockFileTarget> targets = default;
-                PackageSpec lastPackageSpec = default;
-                _lastPackageSpec?.TryGetTarget(out lastPackageSpec);
+                IsInstalledAndTransitiveComputationNeeded = true;
 
-                if ((fileInfo.Exists && fileInfo.LastWriteTimeUtc > _lastTimeAssetsModified) || !ReferenceEquals(lastPackageSpec, packageSpec))
-                {
-                    await TaskScheduler.Default;
-                    if (fileInfo.Exists)
-                    {
-                        var lockFile = LockFileUtilities.GetLockFile(assetsFilePath, NullLogger.Instance);
-                        if (!(lockFile is null))
-                        {
-                            targets = lockFile.Targets;
-                        }
-                    }
-                    _lastTimeAssetsModified = fileInfo.LastWriteTimeUtc;
-                    _lastPackageSpec = new WeakReference<PackageSpec>(packageSpec);
-
-                    // clear the transitive packages cache, since we don't know when a dependency has been removed
-                    _transitivePackages = new List<(NuGetFramework, Dictionary<string, ProjectInstalledPackage>)>();
-                }
-
-                List<PackageReference> installedPackages = packageSpec
-                   .TargetFrameworks
-                   .SelectMany(f => GetPackageReferencesForFramework(f.Dependencies, f.FrameworkName, _installedPackages, targets))
-                   .GroupBy(p => p.PackageIdentity)
-                   .Select(g => g.OrderBy(p => p.TargetFramework, frameworkSorter).First())
-                   .ToList();
-
-                // get the transitive packages, excluding any already contained in the installed packages
-                List<PackageReference> transitivePackages = packageSpec
-                   .TargetFrameworks
-                   .SelectMany(f => GetTransitivePackageReferencesForFramework(f.FrameworkName, _installedPackages, _transitivePackages, targets))
-                   .GroupBy(p => p.PackageIdentity)
-                   .Select(g => g.OrderBy(p => p.TargetFramework, frameworkSorter).First())
-                   .ToList();
-                return new ProjectPackages(installedPackages, transitivePackages);
+                return new ProjectPackages(Array.Empty<PackageReference>(), Array.Empty<TransitivePackageReference>());
             }
-            else
+
+            IList<LockFileTarget> targetsList = null;
+            if (IsInstalledAndTransitiveComputationNeeded)
             {
-                return new ProjectPackages(Array.Empty<PackageReference>(), Array.Empty<PackageReference>());
+                // clear the transitive packages cache, since we don't know when a dependency has been removed
+                ClearCache();
+                targetsList = await GetTargetsListAsync(token);
```

Comment:
I believe we need else statement for `IsInstalledAndTransitiveComputationNeeded is false`, otherwise we never going to make use of cached value, always end up calculating this. If we really need to calculate this all the time then no need this var `IsInstalledAndTransitiveComputationNeeded`.

Reply:
The list below are always calculated, but, those lists are created from instance variables named _installedPackages, _transtivePackages, TransitiveOriginsCache

Reply:
Since we're setting `targetsList` to null by default in line #229 so it'll remain null if `IsInstalledAndTransitiveComputationNeeded` is `false`, is it correct logic?

Reply:
> Since we're setting targetsList to null by default in line https://github.com/NuGet/NuGet.Client/pull/229 so it'll remain null if IsInstalledAndTransitiveComputationNeeded is false, is it correct logic?

Yes. Installed and Transitive packages accounts for this case when `targetsList = null`.

In case of Transitive Origins computation, all packages should return its transitive origins from previously cached results.

Reply:
Discussed offline. No major changes needed here.